src/plugins/str.rs

```diff
@@ -4,65 +4,70 @@ use nu::{
     ReturnSuccess, ReturnValue, ShellError, Tagged, Value,
 };
 
+enum Action {
+    Downcase,
+    Upcase,
+    ToInteger,
+}
+
 struct Str {
     field: Option<String>,
     error: Option<String>,
-    downcase: bool,
-    upcase: bool,
+    action: Option<Action>,
 }
 
 impl Str {
     fn new() -> Str {
         Str {
             field: None,
             error: None,
-            downcase: false,
-            upcase: false,
+            action: None,
         }
     }
 
-    fn is_valid(&self) -> bool {
-        (self.downcase && !self.upcase) || (!self.downcase && self.upcase)
-    }
-
-    fn log_error(&mut self, message: &str) {
-        self.error = Some(message.to_string());
+    fn apply(&self, input: &str) -> Value {
+        match self.action {
+            Some(Action::Downcase) => Value::string(input.to_ascii_lowercase()),
+            Some(Action::Upcase) => Value::string(input.to_ascii_uppercase()),
+            Some(Action::ToInteger) => match input.trim().parse::<i64>() {
+                Ok(v) => Value::int(v),
+                Err(_) => Value::string(input),
+            },
+            None => Value::string(input.to_string()),
+        }
     }
 
     fn for_input(&mut self, field: String) {
         self.field = Some(field);
     }
 
-    fn for_downcase(&mut self) {
-        self.downcase = true;
-
-        if !self.is_valid() {
-            self.log_error("can only apply one")
+    fn update(&mut self) {
```

Comment:
Minor nit: you might want to call this something like `check_if_action_exists`. Update makes it sound like it's updating a value.